### ISSUE TITLE
Rails & Trains

### ISSUE THREAD

**Issue by BenCheung0422** [bitsusei/TerraModulus#66]
*Original: https://github.com/MinicraftPlus/minicraft-plus-revived/issues/287, MinicraftPlus/minicraft-plus-revived/#716 ([deleted](https://gist.github.com/BenCheung0422/46b3299ec1ebc0ab8223a0b3b380f6d6#716-automatic-travel-system))*

## Background

As a part of the Technology system started by #52, this system can be added into the game as a late game feature, separated from the main game flow. These vehicles are more advanced from ridable mobs and thus more powerful comparably.

## Automatic Travel System

This system is quite apart from mob travelling. Anyway, at least there would be trains speaking of the automatic logistics system. The point here is still all those still consume certain fuels or electricity, but more durable than mobs, while mobs may be hungry (depending on implementation) and with weaker health. One more restriction may be that, automatic vehicles cannot jump while mobs may jump over some obstacles. Still, air vehicles would be hard implemented into the game, so it will not be something in consideration. In addition, this system would still depend on #39 for general entity functionality support. Since the vehicles may be powered and programmed, they may still move by themselves even they are not mobs (with AI).

## Details

For a kind of automatic logistics system, trains could be programmed for carrying items, fluids and so on. However, this would rely on an advanced logistics system to be implemented in the game. Since there would be a fixed route indicated by the rails, trains could move just along the routes without being interfered by other parts of the world. Certain mechanisms related to trains would have to be implemented, like the train control logics, movement controls on a fixed area, different rail shapes, storage spaces, etc. The implementation may be referred from [Immersive Railroading](https://www.curseforge.com/minecraft/mc-mods/immersive-railroading) as an example.

## See Also

- #39
- #52

**Comment by BenCheung0422** [comment]
Obsoleted: Project direction changed; Gameplay Contents are not part of the core project
